include/psa/crypto_values.h

```diff
@@ -1487,6 +1487,23 @@
      PSA_ALG_IS_ECDSA(alg) || PSA_ALG_IS_HASH_EDDSA(alg) ||             \
      PSA_ALG_IS_VENDOR_HASH_AND_SIGN(alg))
 
+
+/** Whether the specified algorithm is a signature algorithm that can be used
+ * with psa_sign_message() and psa_verify_message().
+ *
+ * \param alg An algorithm identifier (value of type #psa_algorithm_t).
+ *
+ * \return 1 if alg is a signature algorithm that can be used to sign a
+ *         message. 0 if alg is a signature algorithm that can only be used
+ *         to sign an already-calculated hash. 0 if alg is not a signature
+ *         algorithm. This macro can return either 0 or 1 if alg is not a
+ *         supported algorithm identifier.
+ */
+#define PSA_ALG_IS_SIGN_MESSAGE(alg)                                    \
```

Comment:
Rather than repeat `PSA_ALG_IS_HASH_AND_SIGN`, call it. And add `|| (alg) == PSA_ALG_PURE_EDDSA` which is missing.